_episodes/20-architecture.md

```diff
@@ -1,7 +1,34 @@
 ---
 title: "HPC architecture"
 questions:
-- "To be filled with a question!"
+- "What is a high-performance computer?!"
+- "How are high-performance computers different from personal computers?"
+- "How do these differences influence how I use HPC systems most effectively?"
+objectives:
+- "Provide a general mind map of how HPC resources are connected together (head node, compute node, NAS)"
 keypoints: 
-- "To be filled with a fact/sentence."
+- "A high-performance computer system provides a larger compute capability than is possible to package in a personal computer."
+- "HPC systems are typically an aggregation of a bunch computers, each one of which can look pretty similar to your personal computer."
+- "HPC systems are usually accessed remotely, over the network."
+- "HPC systems are usually shared among many users.  Each user typically gets a dedicated portion of the computer's resources for a period of time."
+- "Special measures have to be taken to provide a file system that can keep up with an HPC system."
+- "HPC systems often provide a lot of different software packages, and provide ways of selecting and configuring them to get the environment you need."
 ---
+
+# What is a High-Performance Computer?
+
+A high-performance computer (HPC system) is a tool used by computational scientists and engineers to tackle problems that require more computing resources or time than they can obtain on the personal computers available to them. HPC systems range in size from the equivalent of just a few personal computers to tens, or even hundreds of thousands of them.  They tend to be expensive to buy and operate, so they are often shared at the departmental or institutional level.  There are also many regional and national HPC centers.  Because of this, most HPC systems are accessed remotely, over the network.
+
+HPC systems are generally constructed from many individual computers, similar in capability to many personal computers.  Each of these individual computers is often referred to as a **node**. HPC systems often include several different types of nodes, which are specialized for different purposes.  **Head** (or **front-end** or **login**) nodes are where you login to interact with the computer.  **Compute** nodes are where the real computing is done.  **Storage** nodes provide the specialized filesystems used on HPC systems.  Some HPC systems also have **service** nodes, which you don't usually interact with directly, but you will sometimes read about.  These nodes are connected by a network (or interconnect), which is often designed to provide very high performance as well.
```

Comment:
I don't think we need to introduce service/storage nodes.  Too many terms!

Reply:
I debated with myself about service nodes.  Storage, I'm not yet persuaded.  I can see an argument of focusing on the filesystems and not worrying about the hardware behind that.   My feeling was that storage nodes might be something they would see in descriptions of machines at some facilities.  I'm willing to be argued out of this feeling.

Reply:
I kinda agree with @ChristinaLK...from the standpoint of HPC software _use_ and _user experience_, I think the only kinds of nodes users will routinely wind up dealing with in conversation, or instructions or in reading documentation would be the compute (back-end) nodes and the login (front-end) nodes. System architectural documents are probably the only places where other kinds of nodes (storage, service, gateway, etc.) might wind up getting described and so are probably outside the scope of a novice HPC lesson.